### ISSUE TITLE
Incorrect loader chain traversal

### ISSUE THREAD

**Issue by a2800276** [go-openapi/loads#28]
`loaders` is a linked list of functionality to load Specs depending on their format (yaml or json). The chain of loaders is set up in the module's `init` function and traversed in `Spec()` but the traversal seems to be broken:

```
    for l := loaders.Next; l != nil; l = l.Next {    // line 121
                if loaders.Match(specURL.Path) {     // line 122
                        b, err2 := loaders.Fn(path)  // line 123
```

- on line 121 the traversal is initialized by setting `l` to the _second_ loader in the list
- `l` is never used in the body of the for loop
- instead, for each loop iteration the head of the list is used to check for applicability of the loader (122) and to actually attempt loading (123)

I stumbled across this investigating another bug and will submit a fix shortly.

## Aside

There's also a `defaultLoader` variable containing a json loader which is tried after iterating the chain loaders. This bit of functionality is redundant because the json loader is also included in the loader chain (just not used, because the traversal is broken). Because `loader` is private, the default json loader can't be removed so it will always be in the loader chain. I feel there are a number of similar issues that could use a little refactoring, but I didn't want to start meddling around before discussing with the team. 

Do you have any mailing list / forum / channel to discuss issues, especially those that span across your multitude of repositories? Any plans to merge all these tiny repos into  single library? Debugging across multiple repos made things a bit more involved than necessary.

**Comment by fredbi** [comment]
@a2800276 

> Do you have any mailing list / forum / channel to discuss issues, especially those that span across your multitude of repositories? Any plans to merge all these tiny repos into single library? Debugging across multiple repos made things a bit more involved than necessary.

https://slackin.goswagger.io/

**Comment by fredbi** [comment]
fixed by #30
